Parse/src/main/java/com/parse/ParseQuery.java

```diff
@@ -960,33 +960,21 @@ public ParseQuery(ParseQuery<T> query) {
     return ParseUser.getCurrentUserAsync();
   }
 
-  private void checkIfRunning() {
-    checkIfRunning(false);
-  }
-
-  private void checkIfRunning(boolean grabLock) {
-    synchronized (lock) {
-      if (isRunning) {
-        throw new RuntimeException(
-            "This query has an outstanding network connection. You have to wait until it's done.");
-      } else if (grabLock) {
-        isRunning = true;
-        cts = Task.create();
-      }
-    }
-  }
-
   /**
    * Cancels the current network request (if one is running).
    */
   //TODO (grantland): Deprecate and replace with CancellationTokens
```

Comment:
I don’t know what that means.. We should overload the signature of find, findInBackground, count & countInBackground to allow a `CancellationToken` ?